neutron/test/convert_test.cpp

```diff
@@ -143,78 +123,44 @@ TEST_P(ConvertTest, Tof_to_DSpacing) {
-                values[3] * 1e-3);
-    // Spectrum 1
-    EXPECT_NEAR(values[4], 3956.0 / (L / tofs[0]) * lambda_to_d,
-                values[4] * 1e-3);
-    EXPECT_NEAR(values[5], 3956.0 / (L / tofs[1]) * lambda_to_d,
-                values[5] * 1e-3);
-    EXPECT_NEAR(values[6], 3956.0 / (L / tofs[2]) * lambda_to_d,
-                values[6] * 1e-3);
-    EXPECT_NEAR(values[7], 3956.0 / (L / tofs[3]) * lambda_to_d,
-                values[7] * 1e-3);
-
-    const auto &data = dspacing["counts"];
-    ASSERT_EQ(data.dims(),
-              Dimensions({{Dim::Spectrum, 2}, {Dim::DSpacing, 3}}));
-    EXPECT_TRUE(equals(data.values<double>(), {1, 2, 3, 4, 5, 6}));
-    EXPECT_EQ(data.unit(), units::counts);
-    ASSERT_EQ(dspacing["counts"].coords()[Dim("position")],
-              tof.coords()[Dim("position")]);
-  } else {
-    ASSERT_TRUE(dspacing.contains("events"));
-    const auto &events = dspacing["events"];
-    ASSERT_EQ(events.dims(), Dimensions(Dim::Spectrum, 2));
-    const auto &tof0 =
-        tof["events"].coords()[Dim::Tof].values<event_list<double>>()[0];
-    const auto &d0 =
-        events.coords()[Dim::DSpacing].values<event_list<double>>()[0];
-    ASSERT_EQ(scipp::size(d0), 4);
-    EXPECT_NEAR(d0[0], 3956.0 / (1e6 * 11.0 / tof0[0]) / sqrt(2.0),
-                d0[0] * 1e-3);
-    EXPECT_NEAR(d0[1], 3956.0 / (1e6 * 11.0 / tof0[1]) / sqrt(2.0),
-                d0[1] * 1e-3);
-    EXPECT_NEAR(d0[2], 3956.0 / (1e6 * 11.0 / tof0[2]) / sqrt(2.0),
-                d0[2] * 1e-3);
-    EXPECT_NEAR(d0[3], 3956.0 / (1e6 * 11.0 / tof0[3]) / sqrt(2.0),
-                d0[3] * 1e-3);
-    const auto &tof1 =
-        tof["events"].coords()[Dim::Tof].values<event_list<double>>()[1];
-    const auto &d1 =
-        events.coords()[Dim::DSpacing].values<event_list<double>>()[1];
-    ASSERT_EQ(scipp::size(d1), 3);
-    EXPECT_NEAR(d1[0], 3956.0 / (1e6 * 11.0 / tof1[0]) * lambda_to_d,
-                d1[0] * 1e-3);
-    EXPECT_NEAR(d1[1], 3956.0 / (1e6 * 11.0 / tof1[1]) * lambda_to_d,
-                d1[1] * 1e-3);
-    EXPECT_NEAR(d1[2], 3956.0 / (1e6 * 11.0 / tof1[2]) * lambda_to_d,
-                d1[2] * 1e-3);
-    ASSERT_EQ(dspacing["events"].coords()[Dim("position")],
-              tof.coords()[Dim("position")]);
-  }
+  EXPECT_EQ(dspacing["counts"].dims(),
+            Dimensions({{Dim::Spectrum, 2}, {Dim::DSpacing, 3}}));
+  // Due to conversion, the coordinate now also depends on Dim::Spectrum.
+  ASSERT_EQ(coord.dims(), Dimensions({{Dim::Spectrum, 2}, {Dim::DSpacing, 4}}));
+  EXPECT_EQ(coord.unit(), units::angstrom);
+
+  const auto values = coord.values<double>();
+  // Rule of thumb (https://www.psi.ch/niag/neutron-physics):
+  // v [m/s] = 3956 / \lambda [ Angstrom ]
+  Variable tof_in_seconds = tof.coords()[Dim::Tof] * (1e-6 * units::one);
+  const auto tofs = tof_in_seconds.values<double>();
+  // Spectrum 0 is 11 m from source
+  // 2d sin(theta) = n \lambda
+  // theta = 45 deg => d = lambda / (2 * 1 / sqrt(2))
```

Comment:
Define a common factor once?
e.g.
```
fact = 3956.0 / (11.0 * sqrt(2.0))
EXPECT_NEAR(values[0], fact * tofs[0], values[0] * 1e-3);
```

Reply:
Unrelated to this PR, so I suggest keeping it.